src/libaktualizr/aktualizr_api.h

```diff
@@ -0,0 +1,241 @@
+/**
+\file
+
+Externally facing API for Aktualizr.
+
+Aktualizr is designed to be run as a standalone component on an embedded system and manage the entire software update
+process. This is suitable for simple cases, but falls short in more complex environments. For example:
+
+- User consent for updates
+- Interactive feedback of installation progress
+
+
+Some of these are not always necessary. Many users today are satisified with applications (Google Docs/GMail, Office
+365, Chrome) that automatically update without user interaction. For these products the designer takes on the
+responsibility of making updates seamless, both in terms of zero downtime for the user and that there are no significant
+regressions. That is not always the case, and Aktualizr is designed to also work in constrained environments where
+updates may make a vehicle undrivable for an extended period of time.
+
+Functional Requirements
+=======================
+
+1. Integration with 3rd party HMI
+2. Integration with 3rd party interfaces to install software on secondary ECUs
+3. Limit network trafic and installation to specific vehicle states
+4. Progress bars
+
+Non-functional Requirements
+===========================
+
+1. Since these interfaces are often proprietry and under NDA, the implementation must be kept separate from the
+Aktualizr tree.
+2. It should be possible to have the integration performed by a separate team
+3. Flexibility and simple integration into new environments.
+
+Overview
+========
+
+The Aktualizr performs all its operations in a background thread, and all the operations except aktualizr_init() and
+aktualizr_free() are asynchronous. Notifications occur via callbacks. These calls are made synchronously from
+Aktualizr's background thread, so they should complete quickly. Normally the callback will only post a notification the
+application's main event loop.
+
+
+Rejected Alternatives
+=====================
+
+It would be possible to allow other components to interface with Aktualizr via
+a IPC interface such as D-Bus or Sockets. This would have the some advantages,
+such as:
+
+ - Process Isolation
+ - Easier to assemble pieces together and hack together things with scripts.
+ - Easier privsep
+
+However we believe the disadvantages outweigh them:
+
+- Have to build an IPC mechanism
+- Have to build bridges to connect this 'common' IPC mechanism to whatever the end-user system has picked
+- New features may require simultanous changes to the aktualizr IPC these bridges.
+- Hard to replace built-in functionality with external functionality (e.g. Package Managers)
+- Harder to completely replace the high-level workflow.
+
+It would still be possible to write a wrapper that would expose libaktualizr
+via an IPC mechanism, but this can be written with a specific use-case and IPC
+technology in mind.
+
+\nosubgrouping
```

Comment:
What is the significance of this "\nosubgrouping"? (And the "\file" at the top?)

Reply:
It appears that `\nosubgrouping` isn't actually needed anymore, I'll remove that.